README.md

```diff
@@ -1,3 +1,133 @@
+indoor differential-drive robot. This robot will have wheel encoders and a horizontal laser.
+
+The first thing we must do is define our state variables. At a minimum, we will want the robot pose at each timestamp.
+Let's assume we model the pose using two different state variables, a 2D position and orientation `(x, y, yaw)`.
+Each 2D pose _at a specific time_ will get a unique variable name. For ease of notation, let's call the pose
+variables `X1`, `X2`, `X3`, etc. In reality, each variable will get a UUID, but those are much harder to write down.
+We will need to derive a 2D pose class from the `fuse_core::Variable` base class. (Actually, `fuse` ships with a 2D
+position and 2D orientation class, but you get the idea. You can create your own variable types when you need them.)
+
+Next we need to decide how to model our sensors. We will probably model the wheel encoders as providing an incremental
+pose measurement. Given a starting pose, `X1` and a wheel encoder delta `z`, you can predict the pose `X2`. So we end
+up with an error term that looks something like `error = f(X1, z)^-1 * X2`. So we need to derive a
+`fuse_core::Constraint` that implements that error function. Similarly, we will probably use some sort of scan-to-scan
+matching scheme using the laser data. This will generate a similar incremental pose measurement.
+
+If our sensors are synchronized, i.e. the laser and the wheel encoders are sampled at the same time, we could construct
+our first `fuse` system at this point. Below is the constraint graph that would be generated. The large circles
+represent state variables at a given time, while the small squares represent measurements. The graph connectivity
+indicates which variables are involved in what measurements.
+
+![fuse graph](doc/fuse_graph_1.png)
+
+However, typically the laser measurements and the wheel encoder measurements are not synchronized. The encoder
+measurements are probably sampled faster than the laser and uses a different clock. In such a situation, the constraint
+graph becomes disconnected.
+
+![fuse graph](doc/fuse_graph_2.png)
+
+This is where motion models come into play. A motion model differs from a sensor model in that constraints can be
+generated between any two requested timestamps. Motion model constraints are generated upon request, not due to their
+own internal clock. We use the motion model to connect the states introduced by the sensor measurements. So we derive
+a class from the `fuse_core::MotionModel` base class that implements our differential drive kinematic constraints.
+
+![fuse graph](doc/fuse_graph_3.png)
+
+Note that there is nothing preventing the system from using multiple motion models at the same time, if that is of
+benefit.
+
+The two sensor models and the motion model are configured as plugins for an optimizer implementation. The optimizer
+handles all of the coordination between the sensors and motion models, as well as performs the required computation
```

Comment:
"as well as _performing_ the"?

Reply:
parallel construction: "handles" and "performs" are the same type of word. If I were a gooder English talking guy, I would tell what kinds of words they were. ("simple present" maybe?)

Reply:
Maybe just replace "as well as" with "and"?

Reply:
Your call, I'm no grammarer. :)